### PR TITLE
Fix for pr of EC_EARTH

### LINKED ISSUE #1111: Dataset problem: non monotinic time coordinate for the pr variable of EC-EARTH
**Describe the dataset issue**

- Data file has been changed by you in any way (if you answer yes the issue will be void and closed, we are not
supporting custom-made datasets that present problems, it is your resposability to fix those problems):
- Project (CMIP5/CMIP6/OBS/obs4mips/etc): CMIP5
- Full dataset description (fill out as many as you know, please):
  - dataset: EC-EARTH
  - experiment: historical
  - mip: Amon
  - variable used: pr
  - data version: v20131231
- Problems encountered (please describe what the actual problems are: e.g. wrong standard name, issue with dimensions):
```
esmvalcore.cmor.check.CMORCheckError: There were errors in variable pr:
Time coordinate for var time is not monotonic
 time: Frequency mon does not match input data
 in cube:
precipitation_flux / (kg m-2 s-1)   (-- : 131; latitude: 160; longitude: 320)
     Dimension coordinates:
          latitude                      -              x               -
          longitude                     -              -               x
     Auxiliary coordinates:
          time                          x              -               -
```
This is because some points in the time coordinate are duplicated and should be removed:
```
cube.coord('time').points[-25:] = masked_array(data=[3242. , 3272.5, 3303.5, 3333. , 3362.5, 3393. , 3423.5,
                   3454. , 3484.5, 3515.5, 3546. , 3576.5, 3607. , 3303.5,
                   3333. , 3362.5, 3393. , 3423.5, 3454. , 3484.5, 3515.5,
                   3546. , 3576.5, 3607. , 3637.5],
             mask=False,
       fill_value=1e+20)
```

- Pointer to existing copy of data on ESGF node (it would be very useful if you could provide a physical
fullpath to the file(s) that are causing the problem, e.g. on CEDA Jasmin or DKRZ):
/mnt/lustre01/work/kd0956/CMIP5/data/cmip5/output1/ICHEC/EC-EARTH/historical/mon/atmos/Amon/r1i1p1/v20131231/pr/pr_Amon_EC-EARTH_historical_r1i1p1_200001-200912.nc
- Other notes and mentions:
Only the time range 2000-2009 is impacted

**Assign the right people**
If you are already a member of the ESMValTool GitHub project, please assign Valeriu Predoi (valeriupredoi) and
Klaus Zimmermann (zklaus) to the issue. They will then check the issue raised and propagate it further to the
data model developers.

**Please attach**
  - The recipe that you are trying to run, you can find a copy in the `run` directory in the output directory
  - The `main_log_debug.txt` file, this can also be found in the `run` directory in the output directory
  - Attach a text file with the output of `ncdump -h /path/to/file.nc`
[recipe_cmip5_EC-EARTH.txt](https://github.com/ESMValGroup/ESMValCore/files/6451695/recipe_cmip5_EC-EARTH.txt)
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/6451696/main_log_debug.txt)

Cheers :beer:

### PR DESCRIPTION
**Pull request by remi-kazeroni**
## Description

Removes erroneously duplicated points in the time coordinate for the last file (time range 2000-2009).

Closes #1111 

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/cmor/_fixes/cmip5/ec_earth.py [hunk 1]
```diff
@@ -129,3 +130,59 @@ def fix_metadata(self, cubes):
...
+                                          var_name=cube.var_name, units=cube.units, attributes=cube.attributes,
+                                          cell_methods=cube.cell_methods,
+                                          dim_coords_and_dims=[(new_time, 0), (lat, 1), (lon, 2)])
+
+                new_list.append(tmp_cube)
+
+            except iris.exceptions.CoordinateNotFoundError:
+                pass
+
+        return new_list
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cmip5/ec_earth.py:188]: @remi-kazeroni a bit of a style not - it's best to have a try/except clause with as little code to try as possible - just one try condition gets executed anyway, and it also looks more readable: try: single clause / except (not clause): raise

Comment (remi-kazeroni) [reply esmvalcore/cmor/_fixes/cmip5/ec_earth.py:205]: @valeriupredoi I have modified the fix to take your suggestion into account. Would that work for you now?

**Comment by remi-kazeroni** [comment]
I mark this PR as RfR even if the tests are failing. I have tried to cover 3 cases:
1. Cubes with non-monotonic time coordinate get fixed
2. Cubes with monotonic time coordinate do not get fixed
3. Cubes with missing time coordinate do not get fixed.
But I'm not sure how to create a "fake time coordinate" with non-monotonic points in iris. @valeriupredoi would you have an idea how to finish the tests?

**Comment by remi-kazeroni** [comment]
Tests are now fixed. @valeriupredoi would you have a moment to review this PR? This is my last IPCC related fix. It would be great if we could have it in the next release but I understand this is short notice 😬

File: esmvalcore/cmor/_fixes/cmip5/ec_earth.py (hunk 1 above)
Comment (valeriupredoi) [reply esmvalcore/cmor/_fixes/cmip5/ec_earth.py:205]: looks good, cheers!

**Review by valeriupredoi — APPROVED** [review]
very cool @remi-kazeroni :beer:

**Pull request merged**